static/js/services/lineage-model-generator.js

```diff
@@ -0,0 +1,49 @@
+angular.module('inboxServices').factory('LineageModelGenerator',
+  function(
+    DB
+  ) {
+    'ngInject';
+    'use strict';
+
+    var get = function(id) {
+      return DB().query('medic-client/docs_by_id_lineage', {
+        startkey: [ id ],
+        endkey: [ id, {} ],
+        include_docs: true
+      });
+    };
+
+    var next = function(result) {
+      var row = result.rows.shift();
+      return row && row.doc;
+    };
+
+    var rest = function(result) {
+      return result.rows.map(function(row) {
+        return row.doc;
+      });
+    };
+
+    var getBaseModel = function(id) {
+      return get(id).then(function(result) {
```

Comment:
what does this do when the `id` is not found? If it returns `{_id: id}` then that's very sneaky, make it throw or return an error.